Assignment 18/classes/AccountTriggerHandler.cls

```diff
@@ -0,0 +1,95 @@
+/**
+* Description   - This trigger helper class is used to create account and send email notification to the customer after account creation.   
+* Created by    - Prateek Bhadoria
+* Created Date  - 27/07/2022
+* Revision log  -  V_1.0 - Created  - Prateek Bhadoria - 27/07/2022
+**/
+public class AccountTriggerHandler {
+    /**   Description :   Create account number as per the account type.
+    *     Created By  :   Prateek Bhadoria
+    *     Arguments   :   
+    *     Return      :   void
+    *     Revision log:   V_1.0 - Created  - Prateek Bhadoria - 27/07/2022
+    *                      
+    **/
+    public static void accountNumberType(List<Account> newAccountList, Map<Id,Account> oldAccountMap) {
+        
+        // Creating 10 digit random number as account number
+        final String strNumber = '0123456789';
+        String accNumber = '';
+        while (accNumber.length() < 10) {
+            Integer createRandomNumber = Math.mod(Math.abs(Crypto.getRandomInteger()), strNumber.length());
+            accNumber += strNumber.substring(createRandomNumber, createRandomNumber+1);
+        }
+        // Iterating Loop to create account number as per the account type
+        for(Account account : newAccountList){
+            if(account.AccountNumber == null){
+                account.AccountNumber = String.valueof(accNumber);
+            }
```

Comment:
Add global if here for (oldAccountMap == null || account.Type != oldAccountMap.get(account.Id).Type)